extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/KubernetesClientProcessor.java

```diff
@@ -58,7 +61,13 @@ public void process(ApplicationIndexBuildItem applicationIndex, CombinedIndexBui
                             watchedClasses.add(watcherGenericTypes.get(0).name().toString());
                         }
                     } catch (IllegalStateException ignored) {
-                    } // no need to handle cases when the generic types could not be determined
+                        // when the class has no subclasses and we were not able to determine the generic types, it's likely that
+                        // the watcher will fail due to not being able to deserialize the class
+                        if (applicationIndex.getIndex().getAllKnownSubclasses(c.name()).isEmpty()) {
+                            log.warn("Watcher '" + c.name() + "' will most likely not work correctly in native mode. " +
```

Comment:
Point to the documentation for more details?

Reply:
Great idea!